### PR TITLE
Add interval

### PR DESCRIPTION
**Pull request by osheamonn**

### CODE REVIEW

File: tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp [hunk 1]
```diff
@@ -12,6 +12,26 @@
...
+ * downcasting `map_base` and then comparing to `map`. Returns false if the
+ * downcast fails.
+ */
+template <typename Map>
+bool are_maps_equal(
+    const Map& map,
+    const CoordinateMapBase<Frame::Logical, Frame::Inertial, Map::dim>&
+        map_base) {
+  try {
+    const Map map_derived = dynamic_cast<const Map&>(map_base);
```
Comment (wthrowe) [inline tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp]: reference

File: tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp [hunk 2]
```diff
@@ -12,6 +12,26 @@
...
+    return map_derived == map;
+  }
+  catch(const std::bad_cast& e){
```
Comment (wthrowe) [inline tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp]: `e` unused (not sure why clang-tidy didn't catch that)

File: src/Domain/DomainCreators/DomainCreator.hpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+//                DomainCreators::RotatedBricks, DomainCreators::UnevenBricks>;
+// };
+}  // namespace DomainCreators_detail
+
+/// Base class for creating Domains from an option string.
+template <size_t VolumeDim, typename TargetFrame>
+class DomainCreator : public Factory<DomainCreator<VolumeDim, TargetFrame>>,
+                      public PUP::able {
+ public:
+  static std::string class_id();
```
Comment (wthrowe) [inline src/Domain/DomainCreators/DomainCreator.hpp]: Remove

File: src/Domain/DomainCreators/DomainCreator.hpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+//       typelist<DomainCreators::Brick, DomainCreators::RubiksCubeWithHole,
+//                DomainCreators::Shell, DomainCreators::Sphere,
+//                DomainCreators::RotatedBricks, DomainCreators::UnevenBricks>;
+// };
+}  // namespace DomainCreators_detail
+
+/// Base class for creating Domains from an option string.
+template <size_t VolumeDim, typename TargetFrame>
+class DomainCreator : public Factory<DomainCreator<VolumeDim, TargetFrame>>,
+                      public PUP::able {
```
Comment (wthrowe) [inline src/Domain/DomainCreators/DomainCreator.hpp]: I believe we decided DomainCreators would not be pupable.

File: src/Domain/DomainCreators/DomainCreator.hpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  DomainCreator() = default;
+  DomainCreator(const DomainCreator<VolumeDim, TargetFrame>&) = delete;
+  DomainCreator(DomainCreator<VolumeDim, TargetFrame>&&) noexcept = default;
+  DomainCreator<VolumeDim, TargetFrame>& operator=(
+      const DomainCreator<VolumeDim, TargetFrame>&) = delete;
+  DomainCreator<VolumeDim, TargetFrame>& operator=(
+      DomainCreator<VolumeDim, TargetFrame>&&) noexcept = default;
+  ~DomainCreator() override = default;
+
+  virtual std::unique_ptr<Domain<VolumeDim, TargetFrame>> create_domain()
```
Comment (wthrowe) [inline src/Domain/DomainCreators/DomainCreator.hpp]: Domain isn't abstract, so no need for the unique_ptr.

File: tests/Unit/Domain/CMakeLists.txt [hunk 6]
```diff
@@ -2,6 +2,7 @@
 # See LICENSE.txt for details.
 
 set(DOMAIN_TESTS
+    Domain/DomainCreators/Test_Interval.cpp
```
Comment (wthrowe) [inline tests/Unit/Domain/CMakeLists.txt]: Alphabetize.

File: tests/Unit/Domain/TestDomainHelpers.hpp [hunk 7]
```diff
@@ -0,0 +1,27 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines helper functions for testing Domain and DomainCreators.
+
+#pragma once
+
+#include "Domain/Domain.hpp"
```
Comment (kidder) [inline tests/Unit/Domain/TestDomainHelpers.hpp]: #include all the stuff

File: tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp [hunk 8]
```diff
@@ -12,6 +12,26 @@
...
+ * \brief Given a Map and a CoordinateMapBase, checks that the maps are equal by
+ * downcasting `map_base` and then comparing to `map`. Returns false if the
+ * downcast fails.
+ */
+template <typename Map>
+bool are_maps_equal(
+    const Map& map,
+    const CoordinateMapBase<Frame::Logical, Frame::Inertial, Map::dim>&
+        map_base) {
+  try {
```
Comment (kidder) [inline tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp]: replace try-catch to checking a nullptr

**Review by kidder — CHANGES_REQUESTED** [review]

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**